File: php-poo/aulas/aula08/Luta.php
<?php
require_once 'Lutador.php';
class Luta
{
    // Atributos
    public $desafiado;
    private $desafiante;
    private $rounds;
    private $aprovada;

    // Métodos Públicos
    public function marcarLuta($l1, $l2) {
        // TODO implement method - √
        if ( $l1->getCategoria() === $l2->getCategoria()
            && ($l1 != $l2) )
        {
            $this->aprovada = true;
            $this->desafiado = $l1;
            $this->desafiante = $l2;
        } else {
            $this->aprovada = false;
            $this->desafiado =  null; // O valor especial null representa uma variável sem valor.
            $this->desafiante = null;
        }
    }
    public function lutar() {
        // TODO implement method
        if ($this->aprovada) {

            $this->desafiado->apresentar();
            $this->desafiante->apresentar();

            $vencedor = rand(0,2);
            // 0-> empate;
            // 1 -> vitoria do desafiado;
            // 2 -> vitoria do desafiante;

            echo "<br>";
            $tracejadoInicial = str_repeat("-", 40);
            echo($tracejadoInicial);

            switch ($vencedor) {
                case 0: // TODO - EMPATE
                    echo "<br><strong> EMPATE! </strong><br>";
                    $this->desafiado->empatarLuta();
                    $this->desafiante->empatarLuta();
                    break;
                case 1: // TODO - GANHOU O DESAFIADO
                    echo "<br><strong> GANHOU A LUTA: " .$this->desafiado->getNome() ."</strong><br>";
                    $this->desafiado->ganharLuta();
                    $this->desafiante->perderLuta();
                    break;
                case 2 : // TODO - GANHOU O DESAFIANTE
                    echo "<br><strong> GANHOU A LUTA: " .$this->desafiante->getNome() ."</strong><br>";
                    $this->desafiante->ganharLuta();
                    $this->desafiado->perderLuta();
                    break;
                }

        } else {
            echo "<br>";
            $tracejadoInicial = str_repeat("-", 40);
            echo($tracejadoInicial);

            echo "<br><span class='foco'> LUTA NÃO PODE ACONTECER! </span> <br>";
        }

        $tracejadoFinal = str_repeat("-", 40);
        echo($tracejadoFinal);
        echo "<br>";
    }

    // Métodos Especiais (Getters e Setters)
    public function getDesafiado()
    {
        return $this->desafiado;
    }

    public function setDesafiado($desafiado): void
    {
        $this->desafiado = $desafiado;
    }

    public function getDesafiante()
    {
        return $this->desafiante;
    }

    public function setDesafiante($desafiante): void
    {
        $this->desafiante = $desafiante;
    }

    public function getRounds()
    {
        return $this->rounds;
    }

    public function setRounds($rounds): void
    {
        $this->rounds = $rounds;
    }

    public function getAprovada()
    {
        return $this->aprovada;
    }

    public function setAprovada($aprovada): void
    {
        $this->aprovada = $aprovada;
    }


}

File: java-poo/projects/poo-aula08/src/aula08/Luta.java
package aula08;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Random;

public class Luta {

    // ATTRIBUTES
    private Lutador desafiado;
    private Lutador desafiante;
    private  int rounds;
    private boolean aprovada;

    // PUBLIC METHODS

    public void mensagem() {
        Date data = new Date();
        SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
        System.out.println("Data: " + df.format(data));

        System.out.println("----------------------------------------");
        System.out.println(" 'Ladies and gentlemen, we are live!' ");
        System.out.println(" 'This is the moment you've all been waiting for' ");
        System.out.println(" 'It's time!' ");
        System.out.println("----------------------------------------");
        // System.out.println("\n");
    }

    public void marcarLuta(Lutador L1, Lutador L2) {

        // L1.getCategoria().equals(L2.getCategoria())
      if ( ( L1.getCategoria() == L2.getCategoria()  )  && ( L1 != L2 ) ) {
              this.aprovada = true;
              this.desafiado = L1;
              this.desafiante = L2;
          } else {
              this.aprovada = false;
              this.desafiado = null;
              this.desafiante = null;
          }
    }

    public void lutar() {

        if (this.aprovada) {
            System.out.println("\n ### DESAFIADO ###");
            this.desafiante.apresentar();
            System.out.println("\n ### DESAFIANTE ###");
            this.desafiado.apresentar();

            // Gerar Numero Aleatorio de 0 a 2, ou seja: { 0 , 1 , 2 }
            Random aleatorio = new Random();

            int vencedor = aleatorio.nextInt(3); // Vai gerar 3 resultados: { 0 , 1 , 2 }

            // 0 = EMPATE
            // 1 = GANHOU DESAFIADO
            // 2 = GANHOU DESAFIANTE"

            switch (vencedor) {
                case 0:
                    System.out.println("\n ### EMPATE: " + this.desafiado.getNome() + " VS "  + this.desafiante.getNome());
                    this.desafiado.empatarLuta();
                    this.desafiante.empatarLuta();
                    break;
                case 1:
                    System.out.println("\n ### GANHOU DESAFIADO! -> " + this.desafiado.getNome());
                    this.desafiado.ganharLuta();
                    this.desafiante.perderLuta();
                    break;
                case 2:
                    System.out.println("\n ### GANHOU DESAFIANTE! -> " + this.desafiante.getNome());
                    this.desafiante.ganharLuta();
                    this.desafiado.perderLuta();
                    break;
                default:
                    System.out.println("NÃO HOUVE VENCEDOR!"); // Opcional
            }
        } else {
            System.out.println(" A luta não pode acontecer!");
        }
    }

    // SPECIAL METHODS
    public Lutador getDesafiado() {
        return desafiado;
    }

    public void setDesafiado(Lutador desafiado) {
        this.desafiado = desafiado;
    }

    public Lutador getDesafiante() {
        return desafiante;
    }

    public void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }

    public int getRounds() {
        return rounds;
    }

    public void setRounds(int rounds) {
        this.rounds = rounds;
    }

    public boolean isAprovada() {
        return aprovada;
    }

    public void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }
}
